src/XRP/shared/gateway-balances.ts

```diff
@@ -0,0 +1,80 @@
+import { XrpUtils } from 'xpring-common-js'
+import { XrpError, XrpErrorType } from '.'
+import {
+  GatewayBalancesResponse,
+  CurrencyValuePair,
+} from './rippled-json-rpc-schema'
+
+/**
+ * Represents an amount of an Issued Currency.
+ */
+export class IssuedCurrencyValue {
+  readonly currency: string
+
+  readonly value: string
+
+  public constructor(currencyValuePair: CurrencyValuePair) {
+    this.currency = currencyValuePair.currency
+    this.value = currencyValuePair.value
+  }
+}
```

Comment:
do you expect this class to ever be more than a bag of values?

if not, imo you should just do

```
interface IssuedCurrencyValue {
  readonly currency: string
  readonly value: string
}

export function issuedCurrencyValueFromPair({currency, value}) {
  return {
    currency,
    value
  }
}
```

Reply:
sweet yeah way better, and I didn't even need the constructor because this is just adding typing to JSON objects that already exist - no transformations even happening